From 52bf1445d2fa4b95dac4b98e1cb6e9dbb5d5ed7d Mon Sep 17 00:00:00 2001
From: Chen Congwu <congwu.chen@intel.com>
Date: Thu, 29 Oct 2009 17:02:52 +0800
Subject: [PATCH] obex_object_resume: fix the cmd code when delivering event for server case

At server side, it is sending response to client so the opcode is the
response code instead of the real cmd code. Should use cmd field instead.
While at client side, it is sending request to server so the cmd code
should use opcode.
---
 lib/obex_object.c |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/lib/obex_object.c b/lib/obex_object.c
index 1073c97..60b14bf 100644
--- a/lib/obex_object.c
+++ b/lib/obex_object.c
@@ -911,6 +911,7 @@ int obex_object_suspend(obex_object_t *object)
 int obex_object_resume(obex_t *self, obex_object_t *object)
 {
 	int ret;
+	uint8_t cmd;
 
 	if (!object->suspend)
 		return 0;
@@ -920,23 +921,22 @@ int obex_object_resume(obex_t *self, obex_object_t *object)
 	if (object->first_packet_sent && !object->continue_received)
 		return 0;
 
+	cmd = (self->state & MODE_SRV) ? object->cmd :
+						object->opcode & ~OBEX_FINAL;
+
 	ret = obex_object_send(self, object, TRUE, FALSE);
 
 	if (ret < 0) {
-		obex_deliver_event(self, OBEX_EV_LINKERR,
-					object->opcode & ~OBEX_FINAL, 0, TRUE);
+		obex_deliver_event(self, OBEX_EV_LINKERR, cmd, 0, TRUE);
 		return -1;
 	} else if (ret == 0) {
-		obex_deliver_event(self, OBEX_EV_PROGRESS,
-					object->opcode & ~OBEX_FINAL, 0,
-					FALSE);
+		obex_deliver_event(self, OBEX_EV_PROGRESS, cmd, 0, FALSE);
 		object->first_packet_sent = 1;
 		object->continue_received = 0;
 	} else {
 		if (self->state & MODE_SRV) {
 			obex_deliver_event(self, OBEX_EV_REQDONE,
-						object->opcode & ~OBEX_FINAL,
-						0, TRUE);
+							cmd, 0, TRUE);
 			self->state = MODE_SRV | STATE_IDLE;
 			return 0;
 		}
-- 
1.7.0.1

